File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/viewmodels/SWImage.java
package dev.radley.omgstarwars.viewmodels;

import dev.radley.omgstarwars.R;
import dev.radley.omgstarwars.models.CategoryOld;

/**
 * Default image resources for Star Wars cards
 */
public class SWImage {

    public static int getPlaceholderImage(String id) {
        switch (id) {
            case CategoryOld.FILMS:
            case CategoryOld.PEOPLE:
            case CategoryOld.SPECIES:
                return R.drawable.placeholder_tall;

            case CategoryOld.STARSHIPS:
            case CategoryOld.VEHICLES:
                return R.drawable.placeholder_wide;
        }
        // planets & fallback
        return R.drawable.placeholder_square;
    }

    public static int getFallbackImage(String id) {
        switch (id) {
            case CategoryOld.FILMS:
                return R.drawable.placeholder_tall;
            case CategoryOld.PEOPLE:
                return R.drawable.generic_people;
            case CategoryOld.PLANETS:
                return R.drawable.generic_planet;
            case CategoryOld.SPECIES:
                return R.drawable.generic_species;
            case CategoryOld.STARSHIPS:
                return R.drawable.generic_starship;
            case CategoryOld.VEHICLES:
                return R.drawable.generic_vehicle;
        }
        // fallback
        return R.drawable.placeholder_square;
    }
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/viewmodels/SWImage.kt
package dev.radley.omgstarwars.viewmodels

import dev.radley.omgstarwars.R
import dev.radley.omgstarwars.models.Category

/**
 * Default image resources for Star Wars cards
 */
object SWImage {

    fun getPlaceholderImage(id: String): Int {
        when (id) {
            Category.FILMS, Category.PEOPLE, Category.SPECIES -> return R.drawable.placeholder_tall

            Category.STARSHIPS, Category.VEHICLES -> return R.drawable.placeholder_wide
        }
        // planets & fallback
        return R.drawable.placeholder_square
    }

    fun getFallbackImage(id: String): Int {
        when (id) {
            Category.FILMS -> return R.drawable.placeholder_tall
            Category.PEOPLE -> return R.drawable.generic_people
            Category.PLANETS -> return R.drawable.generic_planet
            Category.SPECIES -> return R.drawable.generic_species
            Category.STARSHIPS -> return R.drawable.generic_starship
            Category.VEHICLES -> return R.drawable.generic_vehicle
        }
        // fallback
        return R.drawable.placeholder_square
    }
}
